### ISSUE TITLE
[BUG] Missing route/request interception in Firefox when req payload is >=1MB

### ISSUE THREAD

**Issue by ppath** [microsoft/playwright#22753]
### System info
- Playwright Version: v1.33.0
- Operating System: Ubuntu 22.04
- Browser: Firefox
- Other info:

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

https://github.com/ppath/playwright-firefox-buggy-large-req-payload

**Steps**
- npm install
- npm run test

**Expected**

For each test, the [route/request interception](https://github.com/ppath/playwright-firefox-buggy-large-req-payload/blob/main/tests/request_interception.spec.ts#L11) should occur. Therefore, the message `REQUEST INTERCEPTED: http://localhost:3000/api/foo <METHOD>` should be printed three times:
```
  ✓  1 [firefox] › tests/request_interception.spec.ts:17:5 › should intercept POSTing of 10KB payload (1.7s)
REQUEST INTERCEPTED: http://localhost:3000/api/foo POST
[browser]: response payload size: 10260
  ✓  2 [firefox] › tests/request_interception.spec.ts:23:5 › should intercept POSTing of 10MB payload (1.4s)
REQUEST INTERCEPTED: http://localhost:3000/api/foo POST
[browser]: response payload size: 10485780
  ✓  3 [firefox] › tests/request_interception.spec.ts:29:5 › should intercept PUTing of 10MB payload (1.3s)
REQUEST INTERCEPTED: http://localhost:3000/api/foo PUT
[browser]: response payload size: 10485780
```

**Actual**

When the request payload becomes equal or larger than 1MB, the request interception does not occur. Thus, the route handler is executed only once:
```
  ✓  1 [firefox] › tests/request_interception.spec.ts:17:5 › should intercept POSTing of 10KB payload (1.7s)
REQUEST INTERCEPTED: http://localhost:3000/api/foo POST
[browser]: response payload size: 10260
  ✓  2 [firefox] › tests/request_interception.spec.ts:23:5 › should intercept POSTing of 10MB payload (1.4s)
[browser]: response payload size: 10485780
  ✓  3 [firefox] › tests/request_interception.spec.ts:29:5 › should intercept PUTing of 10MB payload (1.3s)
[browser]: response payload size: 10485780
```

So in conclusion, as far as I understand, the problem appears when:
- The request payload becomes >= 1MB (I've tested it with payload of size 1023KB and it worked fine). 
- Running the tests with Firefox.

**Comment by yury-s** [comment]
Thanks for the great repro! I can reproduce it on macOS.
